### PR TITLE
352 practice app frontend

### LINKED ISSUE #128: Complete frontend.py and layout with includes

### PR DESCRIPTION
**Pull request by utulb**
Frontend.py and layout.html with includes are ready to be merged to the master branch.

### CODE REVIEW

**Review by yagizcolak — APPROVED** [review]
Flask code for the frontend and layout work just fine. It is compatible with the other html codes. Keep up the good work.

File: API/FrontEnd/templates/includes/cards.html [hunk 1]
```diff
@@ -0,0 +1,77 @@
+{% macro cards_w_img(products) %}
+<div class="card-deck row-cols-md-3">
+    
+    {% for product in products %}
+        
+    <div class="mb-3">
+        <div class="card text-center"  style="background-color: #10D48E;">
+            <img src= "{{product.url}}" class="card-img-top" height=400 width=400 alt="...">
+            <div class="card-body">
```
Comment (alibatir) [inline API/FrontEnd/templates/includes/cards.html]:
```suggestion
            <div class="card-body" style=" white-space: nowrap;overflow: hidden; text-overflow: ellipsis;">

```

Comment (alibatir) [reply API/FrontEnd/templates/includes/cards.html]: There is an incompatibility between the cards due to their card-title size.
You can add text-overflow property to card-title in order to eliminate the size differences.
This provides card-title to have only one text line.

File: API/FrontEnd/templates/includes/cards.html [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    <div class="mb-3">
+        <div class="card text-center"  style="background-color: #10D48E;">
+            <img src= "{{product.url}}" class="card-img-top" height=400 width=400 alt="...">
+            <div class="card-body">
+              <h2 class="card-title">{{product.name}}</h2>
+              <hr>
+              <h5 class="card-title">Offered by {{product.seller}} </h5>
+              <h5 class="card-title">for {{product.price.try}}₺  {{product.price.usd}}$  {{product.price.eur}}€</h5>
+              <h6 class="card-title">from {{product.location}}</h6>
+              <p class="card-body" style="color: #132A3A; font-weight: 500; font-size: large;"> Product Details: {{product.description}}</p>
```
Comment (alibatir) [inline API/FrontEnd/templates/includes/cards.html]:
```suggestion
             <p class="card-body" style="color: #132A3A; font-weight: 500; font-size: large; white-space: nowrap;
              overflow: hidden; text-overflow: ellipsis;"> Product Details: {{product.description}}</p>

**Review by alibatir — APPROVED** [review]
Overall, very good.

**Review by AsenaKarolin — APPROVED** [review]
The flask code, the includes and the layout.html are all adequate. However there seems to be a conflict with the master branch, apart from resolving that issue the branch is ready to be merged. Good work!

**Review by memetc — APPROVED** [review]
Exchange prices are provided, exmples look fine. Though the html page for creating new product should be renamed to be compatible with the line 49 "create_product.html".

**Pull request merged**